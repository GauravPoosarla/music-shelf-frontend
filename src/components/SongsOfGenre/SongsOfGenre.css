.songs-genre {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  gap: 50px;
}

.genre-details {
  display: flex;
  flex-direction: row;
  padding-left: 120px;
}

.songs-genre > img {
  height: 120px;
  width: 120px;
  border-radius: 50%;
}

.songs-genre-content {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  gap: 5px;
}