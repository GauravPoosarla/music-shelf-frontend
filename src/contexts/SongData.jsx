import React from 'react';
import { createContext } from 'react';
import { useState } from 'react';

export const SongDataContext = createContext({});

export const SongDataProvider = ({ children }) => {
  const [songs, setSongs] = useState([]);

  return (
    <SongDataContext.Provider value={{ songs, setSongs }}>
      {children}
    </SongDataContext.Provider>
  );
};

export default SongDataProvider;
