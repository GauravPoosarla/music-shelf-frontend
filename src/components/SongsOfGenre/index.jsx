import React from 'react';
import Card from '../Card';
import ButtonGenre from '../ButtonGenre';
import './SongsOfGenre.css';

const SongsOfGenre = (props) => {
  return (
    <div className="songs-genre">
      <div className="genre-details">
        <img src={props.genreImage} alt={props.genre} />
        <ButtonGenre genre={props.genre} />
      </div>
      <div className="songs-genre-content">
        {props.songs.map((song, index) => (
          <Card key={index} song={song} index={index} />
        ))}
      </div>
    </div>
  );
};

export default SongsOfGenre;
