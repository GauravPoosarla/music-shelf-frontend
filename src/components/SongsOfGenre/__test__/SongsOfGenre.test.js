import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { SongDataProvider } from '../../../contexts/SongData';
import SongsOfGenre from '..';

describe('SongsOfGenre', () => {
  it('should render the songs of genre', () => {
    const songsByGenre = {
      Country: [
        {
          id: 'cd3cc1e3-e1e0-4ccd-bc67-747648985838',
          name: 'Lost',
          imageUrl:
            'https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631',
          artist: {
            id: '496b0a85-2bfa-45bc-8d0f-57fe0ce55708',
            name: 'Maroon 5',
          },
          genre: {
            id: '128aa7f8-c943-48ce-b352-7edd26fa4c6e',
            name: 'Pop',
          },
        },
        {
          id: '94af94a3-9d47-4dd8-9b45-1c51a9443e96',
          name: 'Maps',
          imageUrl:
            'https://i.scdn.co/image/ab67616d0000b273442b53773d50e1b5369bb16c',
          artist: {
            id: '496b0a85-2bfa-45bc-8d0f-57fe0ce55708',
            name: 'Maroon 5',
          },
          genre: {
            id: '128aa7f8-c943-48ce-b352-7edd26fa4c6e',
            name: 'Pop',
          },
        },
        {
          id: '8266dac3-b2b1-4551-b2ec-91e260244a51',
          name: 'Payphone',
          imageUrl:
            'https://i.scdn.co/image/ab67616d0000b2733119f490f02fcee6514e8604',
          artist: {
            id: '496b0a85-2bfa-45bc-8d0f-57fe0ce55708',
            name: 'Maroon 5',
          },
          genre: {
            id: '128aa7f8-c943-48ce-b352-7edd26fa4c6e',
            name: 'Pop',
          },
        },
      ],
    };
    const genreImages = {
      Country:
        'https://i.scdn.co/image/ab67706c0000bebbf2b5f2f4f4f4f4f4f4f4f4f4',
    };

    const { container } = render(
      <SongsOfGenre
        key={'Country'}
        genre={'Country'}
        songs={songsByGenre['Country']}
        genreImage={genreImages['Country']}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
