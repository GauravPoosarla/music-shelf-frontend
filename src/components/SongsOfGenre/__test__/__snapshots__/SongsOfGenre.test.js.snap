// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongsOfGenre should render the songs of genre 1`] = `
<div>
  <div
    class="songs-genre"
  >
    <div
      class="genre-details"
    >
      <img
        alt="Country"
        src="https://i.scdn.co/image/ab67706c0000bebbf2b5f2f4f4f4f4f4f4f4f4f4"
      />
      <div
        class="button-genre"
      >
        <button>
          Country
        </button>
      </div>
    </div>
    <div
      class="songs-genre-content"
    >
      <div
        class="card card-dark"
      >
        <img
          alt="song img"
          src="https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631"
        />
        <div
          class="song-user"
        >
          <div
            class="song-info"
          >
            <p
              class="song-name"
            >
              Lost
            </p>
            <p
              class="song-artist"
            >
              Maroon 5
            </p>
          </div>
          <div
            class="like"
          >
            <div>
              <button
                class="btn-like"
              >
                0
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card card-light"
      >
        <img
          alt="song img"
          src="https://i.scdn.co/image/ab67616d0000b273442b53773d50e1b5369bb16c"
        />
        <div
          class="song-user"
        >
          <div
            class="song-info"
          >
            <p
              class="song-name"
            >
              Maps
            </p>
            <p
              class="song-artist"
            >
              Maroon 5
            </p>
          </div>
          <div
            class="like"
          >
            <div>
              <button
                class="btn-like"
              >
                0
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card card-dark"
      >
        <img
          alt="song img"
          src="https://i.scdn.co/image/ab67616d0000b2733119f490f02fcee6514e8604"
        />
        <div
          class="song-user"
        >
          <div
            class="song-info"
          >
            <p
              class="song-name"
            >
              Payphone
            </p>
            <p
              class="song-artist"
            >
              Maroon 5
            </p>
          </div>
          <div
            class="like"
          >
            <div>
              <button
                class="btn-like"
              >
                0
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
